Equilobe.DailyReport.SL/ReportService.cs

```diff
@@ -4,41 +4,123 @@
 using Equilobe.DailyReport.Models.ReportFrame;
 using Equilobe.DailyReport.Models.Storage;
 using System;
+using System.Collections.Generic;
 using System.Linq;
 
 namespace Equilobe.DailyReport.SL
 {
     public class ReportService : IReportService
     {
         public IJiraService JiraService { get; set; }
+        public IDataService DataService { get; set; }
+
+        public JiraRequestContext JiraRequestContext { get; set; }
 
         #region IReportService Implementation
         public void UpdateDashboardData(long instanceId)
         {
-            UpdateAtlassianUsers(instanceId);
+            JiraRequestContext = GetJiraRequestContext(instanceId);
+
+            SyncAtlassianUsers(instanceId);
+            SyncAtlassianWorklogs(instanceId);
         }
         #endregion
 
         #region Helpers
-        private void UpdateAtlassianUsers(long instanceId)
+        private void SyncAtlassianWorklogs(long instanceId)
         {
-            var jiraRequestContext = GetJiraRequestContext(instanceId);
-            var users = JiraService.GetAllUsers(jiraRequestContext)
-                .Select(p => ToAtlassianUser(p, instanceId))
-                .ToList();
+            var lastSync = GetLastSyncDate(instanceId);
+
+            SyncDeletedWorklogs(instanceId, lastSync);
+            SyncUpdatedWorklogs(instanceId, lastSync);
+        }
+
+        private DateTime GetLastSyncDate(long instanceId)
+        {
+            using (var db = new ReportsDb())
+            {
+                var worklogs = db.AtlassianWorklogs.OrderByDescending(p => p.LastSync).FirstOrDefault();
```

Comment:
Why do we have the LastSync date on the worklog? I think it will be best to have it on the instance. Because we do not sync a worklog, we sync all the worklogs for the specific instance.